lib/grammers-client/src/types/attributes.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 - developers of the `grammers` project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+use grammers_tl_types as tl;
+
+pub enum Attribute {
+    Audio {
+        voice: bool,
+        duration: i32,
+        title: Option<String>,
+        performer: Option<String>,
+        waveform: Option<Vec<u8>>,
+    },
+    Video {
+        round_message: bool,
+        supports_streaming: bool,
+        duration: i32,
+        w: i32,
+        h: i32,
+    },
+}
+
+impl From<&Attribute> for tl::enums::DocumentAttribute {
```

Comment:
Perhaps we should consume the attribute instead of taking it by reference?

Reply:
I was thinking of that too, done